app/jobs/prepare_payments_job.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class PreparePaymentsJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    Payment
+      .includes(:student)
+      .in_state(:pending)
+      .find_each do |payment|
+      payment.mark_ready!
+    rescue Statesman::GuardFailedError
+      payment.block!
+    end
+  end
```

Comment:
C'est un poil dur à lire.

Une propal:

```suggestion
  def perform
    pending_payments.find_each do |paiement|
      payment.mark_ready!
    rescue Statesman::GuardFailedError
      payment.block!
    end
  end
  
  def pending_payments
    Payment
      .includes(:student)
      .in_state(:pending)
  end
```

Reply:
je trouve pas que l'indirection apporte beaucoup plus dans l'état actuel du code.

Reply:
C'est l'indentation qui est dur à lire, couplé avec le rescue qui est un tab en moins j'ai mis du temps à comprendre à quel endroit c'était catch.